File: zoreza-pro/app/lib/data/models/corte.dart
/// Corte (weekly settlement) model.
class Corte {
  final String uuid;
  final String clienteId;
  final String? clienteNombre;
  final String weekStart;
  final String weekEnd;
  final String? fechaCorte;
  final double comisionPctUsada;
  final double netoCliente;
  final double pagoCliente;
  final double gananciaDueno;
  final String estado;
  final String? createdBy;
  final String? operadorNombre;
  final String? createdAt;
  final String syncStatus;

  const Corte({
    required this.uuid,
    required this.clienteId,
    this.clienteNombre,
    required this.weekStart,
    required this.weekEnd,
    this.fechaCorte,
    required this.comisionPctUsada,
    required this.netoCliente,
    required this.pagoCliente,
    required this.gananciaDueno,
    required this.estado,
    this.createdBy,
    this.operadorNombre,
    this.createdAt,
    this.syncStatus = 'synced',
  });

  bool get isBorrador => estado == 'BORRADOR';
  bool get isCerrado => estado == 'CERRADO';

  factory Corte.fromJson(Map<String, dynamic> json) => Corte(
        uuid: json['uuid'] as String,
        clienteId: json['cliente_id'] as String,
        clienteNombre: json['cliente_nombre'] as String?,
        weekStart: json['week_start'] as String,
        weekEnd: json['week_end'] as String,
        fechaCorte: json['fecha_corte'] as String?,
        comisionPctUsada: (json['comision_pct_usada'] as num).toDouble(),
        netoCliente: (json['neto_cliente'] as num).toDouble(),
        pagoCliente: (json['pago_cliente'] as num).toDouble(),
        gananciaDueno: (json['ganancia_dueno'] as num).toDouble(),
        estado: json['estado'] as String,
        createdBy: json['created_by'] as String?,
        operadorNombre: json['operador_nombre'] as String?,
        createdAt: json['created_at'] as String?,
      );

  /// From CorteSummary JSON (list endpoint).
  factory Corte.fromSummaryJson(Map<String, dynamic> json) => Corte(
        uuid: json['uuid'] as String,
        clienteId: '',
        clienteNombre: json['cliente_nombre'] as String?,
        weekStart: json['week_start'] as String,
        weekEnd: json['week_end'] as String,
        comisionPctUsada: 0,
        netoCliente: (json['neto_cliente'] as num).toDouble(),
        pagoCliente: (json['pago_cliente'] as num).toDouble(),
        gananciaDueno: (json['ganancia_dueno'] as num).toDouble(),
        estado: json['estado'] as String,
      );

  Map<String, dynamic> toDb() => {
        'uuid': uuid,
        'cliente_id': clienteId,
        'week_start': weekStart,
        'week_end': weekEnd,
        'fecha_corte': fechaCorte,
        'comision_pct_usada': comisionPctUsada,
        'neto_cliente': netoCliente,
        'pago_cliente': pagoCliente,
        'ganancia_dueno': gananciaDueno,
        'estado': estado,
        'created_by': createdBy,
        'created_at': createdAt,
        'sync_status': syncStatus,
      };

  factory Corte.fromDb(Map<String, dynamic> row) => Corte(
        uuid: row['uuid'] as String,
        clienteId: row['cliente_id'] as String,
        weekStart: row['week_start'] as String,
        weekEnd: row['week_end'] as String,
        fechaCorte: row['fecha_corte'] as String?,
        comisionPctUsada: (row['comision_pct_usada'] as num).toDouble(),
        netoCliente: (row['neto_cliente'] as num).toDouble(),
        pagoCliente: (row['pago_cliente'] as num).toDouble(),
        gananciaDueno: (row['ganancia_dueno'] as num).toDouble(),
        estado: row['estado'] as String,
        createdBy: row['created_by'] as String?,
        createdAt: row['created_at'] as String?,
        syncStatus: row['sync_status'] as String? ?? 'synced',
      );
}

File: zoreza-pro/backend/app/models/corte.py
"""Corte model — weekly settlement header."""

from __future__ import annotations

import uuid
from datetime import date, datetime

from sqlalchemy import CheckConstraint, Date, DateTime, Float, ForeignKey, String, UniqueConstraint
from app.db.base import UUIDType
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base


class Corte(Base):
    __tablename__ = "cortes"
    __table_args__ = (
        UniqueConstraint("cliente_id", "week_start", name="uq_corte_cliente_week"),
        CheckConstraint("estado IN ('BORRADOR','CERRADO')", name="ck_cortes_estado"),
    )

    cliente_id: Mapped[uuid.UUID] = mapped_column(
        UUIDType(), ForeignKey("clientes.uuid"), nullable=False
    )
    week_start: Mapped[date] = mapped_column(Date, nullable=False)
    week_end: Mapped[date] = mapped_column(Date, nullable=False)
    fecha_corte: Mapped[date | None] = mapped_column(Date, nullable=True)

    comision_pct_usada: Mapped[float] = mapped_column(Float, default=0.0)
    neto_cliente: Mapped[float] = mapped_column(Float, default=0.0)
    pago_cliente: Mapped[float] = mapped_column(Float, default=0.0)
    ganancia_dueno: Mapped[float] = mapped_column(Float, default=0.0)

    estado: Mapped[str] = mapped_column(String(20), default="BORRADOR", server_default="BORRADOR")
    created_by: Mapped[uuid.UUID | None] = mapped_column(
        UUIDType(), ForeignKey("usuarios.uuid"), nullable=True
    )

    # Relationships
    cliente = relationship("Cliente", lazy="selectin")
    operador = relationship("Usuario", foreign_keys=[created_by], lazy="selectin")
    detalles = relationship(
        "CorteDetalle", back_populates="corte", lazy="selectin", cascade="all, delete-orphan"
    )
